packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -0,0 +1,572 @@
+         */
+        withdrawETH: async (
+            amount: NumberLike,
+            opts?: {
+                recipient?: AddressLike
+                overrides?: Overrides
+            }
+        ): Promise<TransactionRequest> => {
+
+            // check if the withdrawer has the sufficient Ether amount
+            const bridgeTx = await this.l2StandardBridge.populateTransaction.bridgeETH(200_000, [], {
+                value: amount,
+            });
+
+            return bridgeTx;
+        },
+
+        /**
+         * Generates a transaction for withdrawing some ERC20 tokens back to the L1 chain.
+         *
+         * @param l1Token Address of the L1 token.
+         * @param l2Token Address of the L2 token.
+         * @param amount Amount to withdraw.
+         * @param opts Additional options.
+         * @param opts.recipient Optional address to receive the funds on L1. Defaults to sender.
+         * @returns Transaction that can be signed and executed to withdraw the tokens.
+         */
+        withdrawERC20: async (
+            l1Token: AddressLike,
+            l2Token: AddressLike,
+            amount: NumberLike,
+            opts?: {
+                recipient?: AddressLike
+            }
+        ): Promise<TransactionRequest> => {
+
+            const withdrawalTx = await this.l2StandardBridge.populateTransaction.bridgeERC20(
+                l2Token.toString(),
+                l1Token.toString(),
+                amount,
+                200_000,
+                [],
+            );
+            return withdrawalTx
+        },
+
+        /**
+         * Generates a transaction for finalizing the deposit on the L2 chain.
+         *
+         * @param bridgeTx Deposit transaction receipt
+         * @returns Transaction that can be signed and executed to finalizethe deposit on L2.
+         */
+        finalizeDeposit: async (
+            bridgeTx: TransactionReceipt,
+        ): Promise<TransactionRequest> => {
+
+            const depositEvent = this.l1Portal.interface.parseLog(bridgeTx.logs[1]);
+            const despositMessage = {
+                version: 0,
+                nonce: depositEvent.args.nonce,
+                sender: depositEvent.args.sender,
+                target: depositEvent.args.target,
+                value: depositEvent.args.value,
+                gasLimit: depositEvent.args.gasLimit,
+                data: depositEvent.args.data,
+            };
+
+            // Get initial block number
+            const initialBlockNumber = bridgeTx.blockNumber;
+
+
+            // get L1 deposit proof
+            const depositProof = await getDepositProof(
+                this.l1Portal.address,
+                depositEvent.args.depositHash,
+                hexlifyBlockNum(initialBlockNumber),
+                this.l1RPCprovider
+            );
+
+            waitUntilBlockConfirmed(this.l1Oracle, initialBlockNumber)
```

Comment:
i think finalize deposit shouldn't wait for the block to be confirmed. it should just check if `l1Oracle.number >= initialBlockNumber` and throw an error if that's not true.
that way SDK users can implement whatever waiting logic they want